src/ui/connector-view/js/connector-view.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Flocking UI MIDI Port Connector
+ * Copyright 2015-2020, Colin Clark
+ * https://github.com/continuing-creativity/flocking-midi
+ *
+ * Dual licensed under the MIT and GPL Version 2 licenses.
+ */
+
+"use strict";
+
+var flock = fluid.registerNamespace("flock");
+
+fluid.defaults("flock.midi.connectorView", {
+    gradeNames: ["flock.midi.receiver", "fluid.viewComponent"],
+
+    portType: "input",
+
+    preferredPort: undefined,
+
+    components: {
+        midiPortSelector: {
+            type: "flock.midi.portSelectorView",
+            container: "{that}.container",
+            options: {
+                portType: "{connectorView}.options.portType",
+                preferredPort: "{connectorView}.options.preferredPort",
+                events: {
+                    onPortSelected: "{connectorView}.events.onPortSelected"
+                }
+            }
+        },
+
+        connection: {
+            createOnEvent: "onValidPortSelected",
+            type: "flock.midi.connection",
+            options: {
+                openImmediately: true,
+                ports: {
+                    expander: {
+                        funcName: "flock.midi.connectorView.generatePortSpecification",
+                        args: [
+                            "{connectorView}.options.portType",
+                            "{midiPortSelector}.selectBox.model.selection"
+                        ]
+                    }
+                },
+
+                // TODO: These are ultimately midi.connection events.
+                // Is there a better way to distribute listeners from this
```

Comment:
There isn't, although these days I think we are preferring to distribute listeners to {component}.event.thing.fire (as you do for afterConnectionOpen) rather than wholesale events since the latter technique can lead to bookkeeping problems

Reply:
Thanks, that's good to know. What kind of bookkeeping issues can occur?

Reply:
We did talk about this a couple of years ago, and as far as I can reconstruct, the main issue that one might run into is that someone registering two listeners with the same namespace on "the event" at the two different sites would end up displacing themselves, rather than having both being active. There might be some other consequences that don't come to mind at the moment ...

Reply:
That makes sense and, I think as you say, something I've likely experienced in the past.